storage/rocksdb/ha_rocksdb.cc

```diff
@@ -15163,7 +15362,8 @@ int ha_rocksdb::external_lock(THD *const thd, int lock_type) {
         DBUG_RETURN(HA_ERR_UNSUPPORTED);
       }
 
-      if (thd->get_explicit_snapshot()) {
+      if (unlikely(tx->has_explicit_snapshot() ||
```

Comment:
why do we need check both(tx and thd)?

Reply:
The state lifetimes don't fully match but only overlap. For instance,
```
CREATE EXPLICIT ROCKSDB SNAPSHOT;
INSERT INTO T1 VALUES(); # tx state not explicit, thd flag = true
```
And the opposite case:
```
START TRANSACTION WITH SHARED ROCKSDB SNAPSHOT;
INSERT INTO T1 VALUES(); # tx state explicit, thd flag = false
```

The tx state is set during transaction execution. The thd snapshot is set on
CREATE/ATTACH ROCKSDB SNAPSHOT, when there is no tx object.

I'm adding a clarifying comment.